tools/src/test/java/org/cyclonedx/schema/XmlCatalogVerificationTest.java

```diff
@@ -0,0 +1,129 @@
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.TestFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.namespace.NamespaceContext;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.junit.jupiter.api.DynamicTest.dynamicTest;
+
+public class XmlCatalogVerificationTest {
+
+    /**
+     * Tests the XML catalog by parsing the xmlcatalog.xml file and verifying if the XML namespaces
+     * in the referenced XSD schema files match the XML namespaces defined in the xmlcatalog.xml file.
+     *
+     * @return  a list of dynamic tests for each URI in the xmlcatalog.xml file
+     * @throws IOException                  if an I/O error occurs while reading the XML catalog file
+     * @throws ParserConfigurationException if a parser configuration error occurs
+     * @throws SAXException                 if a SAX error occurs while parsing the XML catalog file
+     * @throws XPathExpressionException     if an XPath expression error occurs
+     */
+    @TestFactory
+    public List<DynamicTest> testXmlCatalog() throws IOException, ParserConfigurationException, SAXException, XPathExpressionException {
+        // Define the path to the XML catalog file. This is relative to "${basedir}/../schema" in the pom.xml.
+        String xmlCatalogFilename = "xmlcatalog.xml";
+
+        // Load the XML catalog file from the classpath
+        ClassLoader classLoader = getClass().getClassLoader();
+        InputStream xmlCatalogStream = classLoader.getResourceAsStream(xmlCatalogFilename);
+
+        // Ensure the XML catalog file is found
+        Assertions.assertNotNull(xmlCatalogStream, "XML catalog file not found");
+
+        // Parse the xmlcatalog.xml file
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        factory.setNamespaceAware(true);  // Make the factory namespace-aware
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document xmlCatalogDocument = builder.parse(new InputSource(xmlCatalogStream));
+
+        // Get the XML catalog elements
+        NodeList xmlCatalogElements = xmlCatalogDocument.getDocumentElement().getChildNodes();
+
+        // List to hold dynamic tests
+        List<DynamicTest> dynamicTests = new ArrayList<>();
+
+        // Iterate through the XML catalog elements
+        for (int i = 0; i < xmlCatalogElements.getLength(); i++) {
+            Node xmlCatalogElement = xmlCatalogElements.item(i);
+
```

Comment:
could this be refactored to ?
```java
for (int i = 0; i < xmlCatalogElements.getLength(); i++) {
  Node xmlCatalogElement = xmlCatalogElements.item(i);
  if (xmlCatalogElement.getNodeName().equals("uri")) {
    continue;
  }

  dynamicTests.add(dynamicTest("Testing if URI namespace from the XML catalog: " + uriNameXmlCtlg + " matches the URI namespace from the local XSD file: " + targetNamespace, () -> { 

    /// ... all the other things ...

    // Assert if the targetNamespace from the XSD file matches the uriNameXmlCtlg from the XML catalog
    Assertions.assertEquals(uriNameXmlCtlg, targetNamespace, "The namespace " + uriNameXmlCtlg + " does not match the targetNamespace in file " + xsdLocalFilename);
})

}
```

Reply:
I refactor the file, and fixed the indentation. Could you confirm that this is what you had in mind?

Reply:
i dont see any changes. 


my idea was:  put all the test-relevant things into the closure of 
```
  dynamicTests.add(dynamicTest("Testing if URI namespace from the XML catalog: " + uriNameXmlCtlg + " matches the URI namespace from the local XSD file: " + targetNamespace, () -> { 

// put tyhe actual test functionality in here.


});
```

Reply:
In the picture below, these were the modifications I did. I rebased and squashed the commit, that is why you only see 1 commit.
I'm sorry that this does not match what you have in mind.

![diff-XmlCatalogVerificationTest java_20240612_163737](https://github.com/CycloneDX/specification/assets/102670102/548fff06-7312-415c-967f-f07dd8a76024)

I think I understand now.
It should be enough to put the `Assertions.assertEquals(uriNameXmlCtlg, targetNamespace, ...)` in the `dynamicTests.add(dynamicTest(`. Because the real test is in the `Assertions.assertEquals` statement.

Because if the `dynamicTests.add(dynamicTest(` envelops all the xPath code, then I need to remove `uriNameXmlCtlg` and `xsdLocalFilename` from the `dynamicTests.add(dynamicTest(` message, which could be a valid solution as well. But I like to have verbosity in the test message.

Let me know what you think is best: I pushed the modification. Tests running okay on my env.